### ISSUE TITLE
Feature Request: Option to hide S3 methods from completions

### ISSUE THREAD

**Issue by dgkf** [REditorSupport/languageserver#569]
I often find that S3 methods can clutter the completions list, and think it would be helpful to be able to hide S3 methods in most cases.

For example, given a completion token, `pri`, completions look like:

<img src="https://user-images.githubusercontent.com/18220321/185727939-39c60252-db70-4c83-b2cc-d60391b77206.png" width="300" />

With all of the `print.*` methods listed, it's hard to find any non-`print` functions. 

Instead, it would be nice if these methods were hidden until a root of the token is an S3 generic. With the token `pri`, it could produce:

<img src="https://user-images.githubusercontent.com/18220321/185728075-121982c9-4436-498a-8a99-ab6cb0dcd77f.png" width="300" />

Methods are then included in the completions list once the `.` is added to denote a dispatch type. Once the `.` in `print.` is written, methods are included:

<img src="https://user-images.githubusercontent.com/18220321/185728097-63185425-1e31-4af4-b50d-51e83973cb2b.png" width="300" />

I implemented this behavior [in a branch on my fork](https://github.com/dgkf/languageserver/tree/dgkf-hide-s3-method-completions) (though it's not exposed as an option right now), and would be happy to contribute it if you think it fits the project.

**Comment by renkun-ken** [comment]
I like the idea of hiding S3 methods. I'd suggest that if user type a function like `print.` and it should trigger a completion that contains those S3 methods.

**Comment by dgkf** [comment]
> I like the idea of hiding S3 methods. I'd suggest that if user types a function like print. and it should trigger a completion that contains those S3 methods.

Yep - this is how it should work with this PR. See the last screenshot to see the completions after `print.`.

**Comment by renkun-ken** [comment]
Thanks, I tried it in vscode and it works nicely so far, but typing `print.` will not have completions shown but backspace and then re-type `.` will trigger the completion with S3 methods. I guess it is because no non-S3 functions are in the previous completion list triggered by `p`, and backspace-and-reenter `.` triggers again the completion with S3 methods.

**Comment by renkun-ken** [comment]
Would you like to send a PR?

**Comment by dgkf** [comment]
Sure - I'd like to clean it up a bit, and will be unable to do heads down programming for a week or so but I'll put together a PR once I have a bit of time again.
